### ISSUE TITLE
Select's Value Text can't display dynamically update items

### ISSUE THREAD

**Issue by Reverier-Xu** [chakra-ui/ark#2444]
### Description

Reopened for #2029 , this issue seems not fixed in the latest rc.

### Link to Reproduction (or Detailed Explanation)

https://stackblitz.com/edit/solidjs-templates-numplr?file=package.json,src%2FApp.tsx

### Steps to Reproduce

1. create a signal for items
2. add new item to that signal
3. select that new item

Default Placeholder is shown instead of the new item.

### Ark UI Version

3.0.0-3

### Framework

- [ ] React
- [X] Solid
- [ ] Vue

### Browser

Microsoft Edge

**Comment by Reverier-Xu** [comment]
Sorry for the re-post, I hope this issue could be fixed before 3.0.0 release.

**Comment by cschroeter** [comment]
@Reverier-Xu 

Can you please try 3.0.0-4

**Comment by cschroeter** [comment]
@segunadebayo 

😁 

```ts
  const context: select.Context<T> = mergeProps(
    {
      id: createUniqueId(),
      dir: locale().dir,
      getRootNode: environment().getRootNode,
      'open.controlled': props.open !== undefined,
      get collection() {
        return collection()
      },
    },
    localProps,
  )
  
 ```
 
I used this approach - not sure what to do now

**Comment by segunadebayo** [comment]
@cschroeter. All good.

This is now fixed and available in `3.0.0-4`
